routing/segment.hpp

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include "routing/road_point.hpp"
+
+#include "std/cstdint.hpp"
+#include "std/sstream.hpp"
+#include "std/string.hpp"
+
+namespace routing
+{
+// This is directed road segment used as vertex in index graph.
+//
+// You can imagine the segment as a material arrow.
+// Head of each arrow connected to fletchings of next arrows with invisible links:
+// these are the edges of the graph.
+//
+// Position of the segment is a position of the arrowhead: GetPointId(true).
+// This position used in heuristic and edges weight calculations.
+class Segment final
+{
+public:
+  Segment() = default;
+
+  constexpr Segment(uint32_t featureId, uint32_t segmentIdx, bool forward)
+    : m_featureId(featureId), m_segmentIdx(segmentIdx), m_forward(forward)
+  {
+  }
+
+  uint32_t GetFeatureId() const { return m_featureId; }
+  uint32_t GetSegmentIdx() const { return m_segmentIdx; }
+  bool IsForward() const { return m_forward; }
+
+  uint32_t GetPointId(bool front) const
+  {
+    return m_forward == front ? m_segmentIdx + 1 : m_segmentIdx;
+  }
+
+  RoadPoint GetRoadPoint(bool front) const { return RoadPoint(m_featureId, GetPointId(front)); }
+
+  bool operator<(Segment const & seg) const
+  {
+    if (m_featureId != seg.m_featureId)
```

Comment:
Could you please move this to the cpp file? It doesn't look like inline function.